commons/src/test/java/org/openmrs/eip/camel/OauthProcessorTest.java

```diff
@@ -128,11 +129,61 @@ public void process_shouldCallGetNewTokenAndSetTheHeaderIfEnabledAndTheCachedTok
 		OauthToken newCachedOauthToken = getInternalState(processor, "oauthToken");
 		assertNotNull(newCachedOauthToken);
 		assertEquals(expectedNewToken, newCachedOauthToken.getAccessToken());
-		LocalDateTime testLocalDt = ofEpochSecond(testSeconds + expiresIn).atZone(systemDefault()).toLocalDateTime();
+		LocalDateTime testLocalDt = ofEpochSecond(testSeconds + expiresIn - 10).atZone(systemDefault()).toLocalDateTime();
 		assertEquals(testLocalDt, getInternalState(newCachedOauthToken, "expiryDatetime"));
 		assertEquals(HTTP_AUTH_SCHEME + " " + expectedNewToken, exchange.getIn().getBody());
 	}
 	
+	@Test
+	public void process_shouldCallGetNewTokenAndSetTheHeaderIfEnabledAndTheCachedTokenExpiresIn10seconds() throws Exception {
```

Comment:
How different are these new tests from the existing ones you modified above?

Reply:
@wluyima the above use mocked tokens that are predetermined to return false/false on the `isExpired()` function. For these tests, the tokens are not mocked and have actual expiry time to test against.

Reply:
Using mocked vs non mocked tokens makes no difference in my opinion, we use mock tokens because we want to test just the method logic and not that of the token instance themselves, otherwise the new tests are redundant, if you don't want to use mocks for the existing tests, just refactor them to not use mocks instead of duplicating them.